components/eventing-controller/controllers/backend/reconciler.go

```diff
@@ -329,79 +342,49 @@ func setUpEnvironmentForBEBController(secret *v1.Secret) error {
 	return nil
 }
 
-func (r *Reconciler) UpdateBackendStatus(ctx context.Context, backendType eventingv1alpha1.BackendType, backend *eventingv1alpha1.EventingBackend, publisher *appsv1.Deployment, bebSecret *v1.Secret) error {
-	var publisherReady, subscriptionControllerReady bool
-
+func (r *Reconciler) syncBackendStatus(ctx context.Context, desiredStatus eventingv1alpha1.EventingBackendStatus, backend *eventingv1alpha1.EventingBackend, publisher *appsv1.Deployment) error {
 	currentBackend, err := r.getCurrentBackendCR(ctx)
 	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
 		return errors.Wrapf(err, "get current backend failed")
 	}
-	currentStatus := currentBackend.Status
-
-	desiredStatus := getDefaultBackendStatus()
-	desiredStatus.Backend = backendType
 
 	// When backend creation fails, marking the eventingbackend ready to false
 	if backend == nil {
-		// Applying existing attributes
-		desiredBackend := currentBackend.DeepCopy()
-		desiredBackend.Status = desiredStatus
-
-		if object.Semantic.DeepEqual(&desiredStatus, &currentStatus) {
-			return nil
-		}
-
 		r.namedLogger().Debug("update backend CR status when backend is nil")
-		if err := r.Client.Status().Update(ctx, desiredBackend); err != nil {
-			r.namedLogger().Errorw("update EventingBackend status failed", "error", err)
-			return err
-		}
-		return nil
+		// Do not change the value of backend type if it cannot be changed
+		desiredStatus.Backend = currentBackend.Status.Backend
```

Comment:
This comment `// Do not change the value of backend type if it cannot be changed` conflicts with the above comment `// When backend creation fails, marking the eventingbackend ready to false`

Reply:
Both statements are true. But I need to get rid of the second comment.